### DISCUSSION TITLE
Query Result Sharing: Unable to select Query X in panel (from panel with multiple queries)

### DISCUSSION THREAD

**Discussion by MatinF** [grafana/grafana#46356] [Dashboards, Panels & Visualizations]
I am trying to utilize the [query sharing](https://grafana.com/docs/grafana/next/panels/query-a-data-source/share-query/) method in Grafana to setup all my queries in 1 panel - and then reference these via separate individual panels. Being able to do this is critical for my dashboard performance.

For this case I'm using the SimpleJSON datasource plugin and in my 'Query Panel' I have defined two individual queries. One query should be used by Panel 1 and the other by Panel 2.

However, when I select Panel 1 and set the data source to Dashboard, I am only able to reference both of the queries from my Query Panel - not one or the other. This seems very counter-intuitive, so I am wondering if this is a bug? It seems like it should be straight forward to select only one of the individual queries provided by the Query Panel.

My Query Panel: 

![image](https://user-images.githubusercontent.com/26184427/157423039-2d1aebbc-ec12-4799-8aa0-0f9cce4a593c.png)

Panel 1 (where I cannot select e.g. the query all_variables as I would like to): 
![image](https://user-images.githubusercontent.com/26184427/157423083-cdfd9c3f-b7b1-43c6-95b9-0059a40d4e38.png)

**Comment by Elfo404** [comment]
Hello @MatinF, thanks for reporting this.

The feature is working as intended as it allows to share all the results in a panel, also as described in the docs.
I do agree however that it may be an interesting feature.

I'm converting this to a discussion where we track feature requests.

**Reply by MatinF** [reply]
Thanks for clarifying. It just seemed odd to me as I thought the principal purpose of sharing data across panels would be to enable 'selective fetching' of pre-queried data across panels. 

Is there some alternative route to utilize this, such as implementing a filter on each of the individual panels so that I only display data from query X? I would expect it to be a very common use case that a dashboard should send 1 request to the backend to generate data for use by all panels. In our case, the backend will "process data on demand" upon the request being sent. If we have to send an individual request for each panel, the backend will essentially process all of the data all over again for each request. 

Martin

**Comment by noheton** [comment]
Sorry for being so blunt - has there been any progress on this topic?
Currently we provision (over the API) a preliminary dashboard, where each query has its own panel. 
Now we would like to enable users (who are not familiar with writing queries) to mix and match the queries to get "their" data visualization.
Of course, we could group them before provisioning, but as each user has different visualization preferences, it would require an additional "query to panel grouping"-feature on our part. Doable for sure - but i was hoping for a more integrated solution, as we would like users to work in grafana directly as much as possible.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
